File: leetcode9.go
// +build ignore

package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	y := 0
	t := x
	for 0 != x/10 || 0 != x%10 {
		y = y*10 + x%10
		x = x / 10
	}
	if t == y {
		return true
	}
	return false
}

func main() {

}

File: leetcode9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 :
            return False
        t = x
        y = 0
        while t // 10  or t % 10:
            y = y * 10 + t%10
            t = t // 10
        if x == y:
            return True
        return False